commands/operator-sdk/cmd/test/cluster.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmdtest
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	testNamespace     string
+	kubeconfigCluster string
+	imagePullPolicy   bool
+)
+
+func NewTestClusterCmd() *cobra.Command {
+	testCmd := &cobra.Command{
+		Use:   "cluster <image name> [flags]",
+		Short: "Run End-To-End tests using image with embedded test binary",
+		RunE:  testClusterFunc,
+	}
+	defaultKubeConfig := ""
+	homedir, ok := os.LookupEnv("HOME")
+	if ok {
+		defaultKubeConfig = homedir + "/.kube/config"
+	}
+	testCmd.Flags().StringVarP(&testNamespace, "namespace", "n", "default", "Namespace to run tests in")
+	testCmd.Flags().StringVarP(&kubeconfigCluster, "kubeconfig", "k", defaultKubeConfig, "Kubeconfig path")
+	testCmd.Flags().BoolVarP(&imagePullPolicy, "imagePullPolicy", "i", false, "Set test pod image pull policy to 'Never'")
+
+	return testCmd
+}
+
+func testClusterFunc(cmd *cobra.Command, args []string) error {
+	// in main.go, we catch and print errors, so we don't want cobra to print the error itself
+	cmd.SilenceErrors = true
+	if len(args) != 1 {
+		return fmt.Errorf("operator-sdk test cluster requires exactly 1 argument")
+	}
+	// cobra prints its help message on error; we silence that here because any errors below
+	// are due to the test failing, not incorrect user input
+	cmd.SilenceUsage = true
+	var pullPolicy v1.PullPolicy
+	if imagePullPolicy {
+		pullPolicy = v1.PullNever
+	} else {
+		pullPolicy = v1.PullAlways
```

Comment:
set pullPolicy to always when you declare and remove the else block. I think this is apart of most go style checks that we have not added.

Reply:
Golint doesn't seem to catch that. What linting tool do you think might make sense to implement for this repo in the future? The only real style check we have is capialization of error strings, which we do with a shell script instead of `golint`, which checks a lot more things like comments on all exported functions and variables, which is something we haven't done.